clients/java/src/main/java/marquez/client/MarquezPathV1.java

```diff
@@ -1,76 +1,102 @@
 package marquez.client;
 
 import com.google.common.annotations.VisibleForTesting;
+import java.util.ArrayList;
+import java.util.List;
 import javax.annotation.Nullable;
 import lombok.NonNull;
 import marquez.client.models.RunState;
 
 class MarquezPathV1 {
-
   @VisibleForTesting static final String BASE_PATH = "/api/v1";
 
   @VisibleForTesting
-  static String path(String pathTemplate, @Nullable String... pathArgs) {
-    return BASE_PATH + String.format(pathTemplate, (Object[]) pathArgs);
+  static List<String> path(String pathTemplate, @Nullable String... pathArgs) {
+    /*
+     Converts path template (prepended with BASE_PATH), where path parts are separated
+     by slashes, and formats strings filling pathArgs in place of %s placeholder.
+     This is required to properly handle double slashes in URLs using URIBuilder.
+     Example:
+     Path template "/namespaces/%s/datasets/%s/versions/%s"
+     with args
+     "nName", "dName", "vName"
+     is converted to list
+     ["api", "v1", "namespaces", "nName", "datasets", "dName", "versions", "vName"]
+    */
+    List<String> resultPath = new ArrayList<>();
```

Comment:
I didn't get why do we need that, but what about stream approach?
I have only one concern, what if number of %s and pathArgs mismatch, should we throw some exception, or put empty string instead of %s?
```
    String pathTemplate = "/namespaces/%s/datasets/%s/versions/%s";
    String[] pathArgs = new String[]{"nName", "dName", "vName"};
    Iterator<String> iterator = Arrays.stream(pathArgs).iterator();
    List<String> parts = Stream.of(pathTemplate.split("/"))
            .filter(StringUtils::isNotBlank)
            .map(it->it.equals("%s") ? iterator.next() : it)
            .collect(Collectors.toList());
    assertThat(parts).isEqualTo(Arrays.asList("namespaces", "nName", "datasets", "dName", "versions", "vName"));
```